packages/kyc/src/controllers/address/controller.ts

```diff
@@ -0,0 +1,75 @@
+import { AdapterError, Requester, Validator } from '@chainlink/ea-bootstrap'
+import { AdapterRequest, Config, ExecuteWithConfig } from '@chainlink/types'
+import { selectors } from '@linkpool/shared'
+
+import api from '../../api'
+import { RequestMethod } from '../constants'
+import { getSingleInputParameters, inputParameters } from './input'
+
+const controller: ExecuteWithConfig<Config> = async (request: AdapterRequest) => {
+  const validator = new Validator(request, inputParameters)
+  const client = api()
+
+  const jobRunID = validator.validated.id
+  const method = validator.validated.data.method
+  const parse = validator.validated.data.parse
+
+  let result: Record<string, any> | Record<string, any>[]
+  try {
+    switch (method) {
+      case RequestMethod.GET: {
+        result = (await client.address.get()) as Record<string, any>
```

Comment:
```
// TODO: validate network and address before making the request! otherwise you are wasting resources if the inputs
// are invalid
result = (await client.addresses.get()) as Record<string, any>
```
And then filter by `network` and `lookupAddress` in this switch case. Encapsulate it in a specific utils and don't loop the array twice.